### PR TITLE
Rh del endpoint

### LINKED ISSUE #13: Add `DELETE` endpoint for a specific record in UCSBSubjects table.
Add `DELETE` endpoint for a specific record in UCSBSubjects table.

# Acceptance Criteria:

- [ ] In `UCSBSubjectController.java` there is code for an 
      endpoint `DELETE /api/UCSBSubjects?id=123` endpoint 
      that deletes the record if it exists, and returns 200 (ok) and 
      the text `record 123 deleted`, or returns 400 (Bad Request) and
      the text `record 123 not found` if it does not.
- [ ] The Swagger-UI endpoints for this endpoint is well documented
      so that any member of the team can understand how to use it.
- [ ] The endpoint works as expected on localhost.
- [ ] The endpoint works as expected when deployed to Heroku.
- [ ] There is full test coverage (Jacoco) for the new code in 
      `UCSBSubjectController.java`
- [ ] There is full mutation test coverage (Pitest) for new code in
      `UCSBSubjectController.java`

# Details

Consult the methods in [`TodosController.java`](https://github.com/ucsb-cs156-w22/demo-spring-react-example-v2/blob/main/src/main/java/edu/ucsb/cs156/example/controllers/TodosController.java) for examples.

Consult the tests in [`TodosControllerTests.java`](https://github.com/ucsb-cs156-w22/demo-spring-react-example-v2/blob/main/src/test/java/edu/ucsb/cs156/example/controllers/TodosControllerTests.java) for examples.

### PR DESCRIPTION
**Pull request by bobhe9606**
# Overview

In a sentence or two, provide an overview of how this pull request updates the main branch.  You may use a bullet list if that helps.

# Issues Addressed

If this PR closes or deals with issues, mention them and tag the issue number with the `#` syntax.

# Details

Where applicable, include "before" and "after" screenshots or animated GIFs.   The tool [LiceCAP](https://www.cockos.com/licecap/) or one of [these similar tools](https://www.nextofwindows.com/5-free-tools-to-screen-capture-to-gif-on-windows) 
can be used to capture screenshots/animated GIFs.

## Reminders (delete this section before merging or as code is merged)
- [ ] Explicitly link to this pull request each issue that it closes
- [ ] Assign one and ideally two reviewers to review this pull request. 
- [ ] Reviewers should not just approve with "Looks good!" Instead, reviewers should carefully go through the code and use GitHub's reviewer interface to ask questions, make comments, and suggest changes.

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java [hunk 1]
```diff
@@ -104,19 +106,21 @@ public UCSBSubject postUCSBSubject(
     }
 /*
```
Comment (AHuangHe) [inline src/main/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectController.java]: Try uncommenting this for the test?

**Review by AHuangHe — APPROVED** [review]
LGTM

File: src/test/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectControllerTests.java [hunk 2]
```diff
@@ -139,8 +140,115 @@ public void api_get_id_returns_a_subject_that_does_not_exist() throws Exception
...
+                        .andExpect(status().isOk()).andReturn();
+
+                // assert
+                verify(ucsbsubjectRepository, times(1)).findById(15L);
+                verify(ucsbsubjectRepository, times(1)).deleteById(15L);
+                String responseString = response.getResponse().getContentAsString();
+                assertEquals("ucsb subject with id 15 deleted", responseString);
+        }
+
+
```
Comment (AHuangHe) [inline src/test/java/edu/ucsb/cs156/team02/controllers/UCSBSubjectControllerTests.java:278]: Oh, I think you are missing the test case where the user wants to delete a subject that does not exist

**Review by AHuangHe — CHANGES_REQUESTED** [review]
Needs an additional test case

**Review by edisonnchen — APPROVED** [review]
LGTM

**Pull request merged**